backend/open_webui/models/chats.py

```
@@ -752,73 +1394,96 @@ def get_chat_list_by_user_id_and_tag_name(
+        tag_id = tag_name.replace(" ", "_").lower()
+        Tags.ensure_tags_exist([tag_name], user_id, db=db)
         try:
-            with get_db() as db:
+            with get_db_context(db) as db:
                 chat = db.get(Chat, id)
-
-                tag_id = tag.id
                 if tag_id not in chat.meta.get("tags", []):
                     chat.meta = {
                         **chat.meta,
                         "tags": list(set(chat.meta.get("tags", []) + [tag_id])),
```

Ruff (0.15.4)

[warning] Consider `[*chat.meta.get("tags", []), tag_id]` instead of concatenation

Replace with `[*chat.meta.get("tags", []), tag_id]`

(RUF005)